services/submission/src/main/java/app/coronawarn/server/services/submission/controller/SubmissionController.java

```diff
@@ -145,6 +159,27 @@
     return diagnosisKeys;
   }
 
+  private void checkDiagnosisKeysStructure(List<DiagnosisKey> diagnosisKeys) {
+    diagnosisKeys.sort(Comparator.comparing(DiagnosisKey::getRollingStartIntervalNumber));
+    String keysString = Arrays.toString(diagnosisKeys.toArray());
+    Predicate<DiagnosisKey> hasRiskLevel6 = diagnosisKey -> diagnosisKey.getTransmissionRiskLevel() == 6;
+
+    if (diagnosisKeys.stream().noneMatch(hasRiskLevel6)) {
```

Comment:
the method name says checkDiagnosisKeysStructure which is more general. But in the implementation only risk 6 keys are being checked. Shouldn't we have logging for all types risk levels which would be expected in the payloads but are missing?

Reply:
The number of keys to be submitted depends on the time how long the app is installed on the phone. Hence, the number of keys differs, but the key with transmission risk level 6 is the one from the day before submission and should always be present.

Reply:
I understand. It's worth adding a javadoc with your explanation

Reply:
done ;)